File: src/org/usfirst/frc5293/translations/camera/CameraPipeline.java
package org.usfirst.frc5293.translations.camera;

import org.usfirst.frc5293.translations.util.OperationPipeline;
import org.usfirst.frc5293.translations.util.Point;

public class CameraPipeline extends OperationPipeline<Point> {

    private static CameraPipeline instance;

    public static CameraPipeline getInstance() {
        if (instance == null) {
            instance = new CameraPipeline();
        }

        return instance;
    }

    //

    private final CameraTranslations x = CameraTranslations.getInstance();

    private CameraPipeline() {
        ops.add(x::applyInverting);
        ops.add(x::applyQuadScaling);
        ops.add(x::applyInputScaling);
        ops.add(x::applyOutputLimit);
    }
}

File: src/main/kotlin/org/usfirst/frc5293/translations/camera/CameraPipeline.kt
package org.usfirst.frc5293.translations.camera

import org.usfirst.frc5293.translations.operations.applyInputScaling
import org.usfirst.frc5293.translations.operations.applyInverting
import org.usfirst.frc5293.translations.operations.applyOutputLimit
import org.usfirst.frc5293.translations.operations.applyQuadScaling
import org.usfirst.frc5293.translations.util.OperationPipeline
import org.usfirst.frc5293.translations.util.Point

object CameraPipeline : OperationPipeline<Point> {

    override val operations: List<(Point) -> Point> = listOf(
            ::applyInverting,
            ::applyQuadScaling,
            ::applyInputScaling,
            ::applyOutputLimit
    )
}
